### ISSUE TITLE
cuda::detail::trivial_copy_n needs to use cudaStreamLegacy

### ISSUE THREAD

**Issue by jaredhoberock** [thrust/thrust#685]
The intention is for host <-> device copies to be synchronous.

To implement this, [we've used stream 0 in the past in `trivial_copy_n`](https://github.com/thrust/thrust/blob/master/thrust/system/cuda/detail/trivial_copy.inl#L120).

Since the user is able to change the meaning of stream 0 now, we need to use `cudaStreamLegacy` here to get the same behavior.

**Comment by jaredhoberock** [comment]
Should probably also use `cudaStreamLegacy` in the other place we've used 0; i.e., in [this file](https://github.com/thrust/thrust/blob/master/thrust/system/cuda/detail/execute_on_stream.h).

**Comment by eglaser77** [comment]
Wouldn't it be better to just do a cudaStreamSynchronize after the cudaMemcpyAsync call? By specifying cudaStreamLegacy won't this synchronize all streams?

**Comment by jaredhoberock** [comment]
Yes, it would be better, but the problem is identifying which copies calls require synchronization. #684 has some details.
